src/scaraplate/config.py

```diff
@@ -15,11 +15,34 @@ class StrategyNode(NamedTuple):
     config: Dict[str, Any]
 
 
-class ScaraplateYaml(NamedTuple):
-    default_strategy: StrategyNode
-    strategies_mapping: Mapping[str, StrategyNode]
-    git_remote_type: Optional[Type[GitRemote]]
-    cookiecutter_context_type: Type[CookieCutterContext]
+class ScaraplateYaml:
```

Comment:
`ScaraplateYaml` is sort of a dataclass. Let's keep it as `NamedTuple` for now. Or you see any reasons why it shouldn't be such?

Reply:
@kxepal I believe it was turned to a class in order to allow mutation of the `strategies_mapping` property below.

Although I would prefer to keep it a NamedTuple as well.

Reply:
Indeed I had to turn it into a class for mutating `strategies_mapping`